Sources/SwiftParser/Types.swift

```diff
@@ -900,33 +900,33 @@ extension Parser.Lookahead {
 }
 
 extension Parser {
-  private mutating func parseLifetimeTypeSpecifier(
-    specifierHandle: TokenConsumptionHandle
-  ) -> RawTypeSpecifierListSyntax.Element {
-    let specifier = self.eat(specifierHandle)
+  private mutating func parseLifetimeTypeSpecifier() -> RawTypeSpecifierListSyntax.Element {
+    let specifier = self.eat(TokenSpec(.dependsOn))
```

Comment:
Could you use `self.expect` instead of `self.eat`. That’s generally safer because it won’t hit an assertion failure if we currently aren’t positioned at `dependsOn` (which might become the case in the future if we should decide to change the `self.at(.dependsOn)` to `self.canRecoverTo(.dependsOn)` in `parseTypeAttributeList ` in the future). We just generally try to be as error-tolerant as possible in the parser.

---

Also, I would call the variable `depensOnKeyword` because it no longer is an arbitrary specifier.

---

To be consistent with the rest of the codebase I would use `.keyword(.dependsOn)` instead of `TokenSpec(.dependsOn)`.